### PR TITLE
[DPE-4924] Add release notes to documentation

### PR DESCRIPTION
**Pull request by deusebio**
This PR adds the release notes of previous releases to the official charmhub documentation

@zmraul @Batalex

### CODE REVIEW

**Review by marcoppenheimer — APPROVED** [review]

File: docs/reference/releases/rev156_126.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Revision 156/126
+<sub>Wednesday, February 28, 2024</sub>
+
+Dear community,
+
+We are extremely thrilled and excited to share that Charmed Kafka and Charmed ZooKeeper have now been released as GA. You can find them in [charmhub.io](https://charmhub.io/) under the `3/stable` track.
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:6]:
```suggestion
"We are thrilled to announce that [Charmed Kafka](https://charmhub.io/kafka) and [Charmed ZooKeeper](https://charmhub.io/zookeeper) have reached General Availability (GA). You can find them on [charmhub.io](https://charmhub.io/) under the `3/stable` track."
```

File: docs/reference/releases/rev156_126.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+More information are available in the [Canonical website](https://canonical.com/data/kafka), alongside its [documentation](https://canonical.com/data/docs/kafka/iaas).
+Also find the full announcement of the release [here](https://canonical.com/blog/charmed-kafka-general-availability) and [here](https://discourse.charmhub.io/t/announcing-general-availability-of-charmed-kafka/13277). 
+And more importantly, make sure you don't miss out the [webinar](https://www.linkedin.com/events/7161727829259366401/about/) that Raúl Zamora and Rob Gibbon will be holding later today.
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:10]:
```suggestion
The full version of the announcement can be seen in our [blog](https://canonical.com/blog/charmed-kafka-general-availability) and [forum](https://discourse.charmhub.io/t/announcing-general-availability-of-charmed-kafka/13277). 
For more information on Charmed Kafka, see the [Canonical website](https://canonical.com/data/kafka), [documentation](https://canonical.com/data/docs/kafka/iaas), and our webinar [Introduction to Apache Kafka](https://www.linkedin.com/events/7161727829259366401/about/) by Raúl Zamora and Rob Gibbon.
```
Some suggestions on wording.
Also, the webinar announcement should be more specific on the topic and less bound to the current date (ideally, we want it to stay correct before, during, and after the webinar).

File: docs/reference/releases/rev156_126.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+Please reach out should you have any question, comment, feedback or information. You can find us here in [Matrix](https://matrix.to/#/#charmhub-data-platform:ubuntu.com) or also on [Discourse](https://discourse.charmhub.io/).
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:12]:
```suggestion
Please feel free to reach out to us should you have any questions, comments, or feedback. You can find us here in [Matrix](https://matrix.to/#/#charmhub-data-platform:ubuntu.com) or on [Discourse](https://discourse.charmhub.io/).
```

File: docs/reference/releases/rev156_126.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+## Features you can start using today:
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:14]:
```suggestion
## Features
```

File: docs/reference/releases/rev156_126.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:15]:
```suggestion

Here is the list of features you can start using today:

```

File: docs/reference/releases/rev156_126.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+* Deploying on VM (tested with LXD, MAAS)
+* ZooKeeper using SASL authentication
+* Scaling up/down in one simple juju command
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:18]:
```suggestion
* Scaling up/down in one simple Juju command
```

File: docs/reference/releases/rev156_126.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+* Persistent storage support with Juju Storage
+* Super-user creation
+* Documentation featuring Diàtaxis framework
+
+Canonical Data issues are now public on both [Jira](https://warthogs.atlassian.net/jira/software/c/projects/DPE/issues/) 
+and [GitHub](https://github.com/canonical/kafka-operator/issues) platforms.
+
+[GitHub Releases](https://github.com/canonical/kafka-operator/releases) provide a detailed list of bugfixes, PRs, and commits for each revision.
+
+## Inside the charms:
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:33]:
```suggestion
## Versions
```

File: docs/reference/releases/rev156_126.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+* Charmed ZooKeeper charm ships the ZooKeeper [3.8.2-ubuntu0](https://launchpad.net/zookeeper-releases/3.x/3.8.2-ubuntu0), built and supported by Canonical
+* Charmed Kafka charm ships the Kafka [3.6.0-ubuntu0](https://launchpad.net/kafka-releases/3.x/3.6.0-ubuntu0), built and supported by Canonical
+* Charmed ZooKeeper charm is based on [charmed-zookeeper snap](https://snapcraft.io/charmed-zookeeper) on the `3/stable` (Ubuntu LTS “22.04” - core22-based)
+* Charmed Kafka charm is based on [charmed-kafka snap](https://snapcraft.io/charmed-kafka) on the `3/stable` channel (Ubuntu LTS “22.04” - core22-based)
+* Principal charms supports the latest LTS series “22.04” only.
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:39]:
```suggestion
Principal charms support the latest LTS series “22.04”:

* ZooKeeper charm is based on the [`charmed-zookeeper` snap](https://snapcraft.io/charmed-zookeeper) on the `3/stable` (Ubuntu 22.04 LTS - core22-based) and ships the ZooKeeper [3.8.2-ubuntu0](https://launchpad.net/zookeeper-releases/3.x/3.8.2-ubuntu0), built and supported by Canonical.
* Kafka charm is based on the [`charmed-kafka` snap](https://snapcraft.io/charmed-kafka) on the `3/stable` channel (Ubuntu 22.04 LTS - core22-based) and ships the Kafka [3.6.0-ubuntu0](https://launchpad.net/kafka-releases/3.x/3.6.0-ubuntu0), built and supported by Canonical.
```

File: docs/reference/releases/rev156_126.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+More information about the artifacts are provided by the following table:
+
+| Artifact               | Track/Series | Version/Revision | Code                                                                                                                |
+|------------------------|--------------|------------------|---------------------------------------------------------------------------------------------------------------------|
+| zookeeper distribution | 3.x          | 3.8.2-ubuntu0    | [5bb82d](https://git.launchpad.net/zookeeper-releases/tree/?h=lp-3.8.2&id=5bb82df4ffba910a5b30dd42499921466405f087) |
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:45]:
```suggestion
| ZooKeeper distribution | 3.x          | 3.8.2-ubuntu0    | [5bb82d](https://git.launchpad.net/zookeeper-releases/tree/?h=lp-3.8.2&id=5bb82df4ffba910a5b30dd42499921466405f087) |
```

File: docs/reference/releases/rev156_126.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+| kafka distribution     | 3.x          | 3.6.0-ubuntu0    | [424389](https://git.launchpad.net/kafka-releases/tree/?h=lp-3.6.0&id=424389bb8f230beaef4ccb94aca464b5d22ac310)     |
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:46]:
```suggestion
| Kafka distribution     | 3.x          | 3.6.0-ubuntu0    | [424389](https://git.launchpad.net/kafka-releases/tree/?h=lp-3.6.0&id=424389bb8f230beaef4ccb94aca464b5d22ac310)     |
```

File: docs/reference/releases/rev156_126.md [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+| charmed-zookeeper-snap | 3/stable     | 28               | [9757f4](https://github.com/canonical/charmed-zookeeper-snap/tree/9757f4a2a889981275f8f2a1a87e1c78ae1adb77)         |
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:47]:
```suggestion
| `charmed-zookeeper-snap` | 3/stable     | 28               | [9757f4](https://github.com/canonical/charmed-zookeeper-snap/tree/9757f4a2a889981275f8f2a1a87e1c78ae1adb77)         |        
```

File: docs/reference/releases/rev156_126.md [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+
+More information about the artifacts are provided by the following table:
+
+| Artifact               | Track/Series | Version/Revision | Code                                                                                                                |
+|------------------------|--------------|------------------|---------------------------------------------------------------------------------------------------------------------|
+| zookeeper distribution | 3.x          | 3.8.2-ubuntu0    | [5bb82d](https://git.launchpad.net/zookeeper-releases/tree/?h=lp-3.8.2&id=5bb82df4ffba910a5b30dd42499921466405f087) |
+| kafka distribution     | 3.x          | 3.6.0-ubuntu0    | [424389](https://git.launchpad.net/kafka-releases/tree/?h=lp-3.6.0&id=424389bb8f230beaef4ccb94aca464b5d22ac310)     |
+| charmed-zookeeper-snap | 3/stable     | 28               | [9757f4](https://github.com/canonical/charmed-zookeeper-snap/tree/9757f4a2a889981275f8f2a1a87e1c78ae1adb77)         |        
+| zookeeper-operator     | 3/stable     | 126              | [9ebd9a](https://github.com/canonical/zookeeper-operator/commit/9ebd9a2050e0bd626feb0019222d45f211ca7774)           |
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:48]:
```suggestion
| `zookeeper-operator`     | 3/stable     | 126              | [9ebd9a](https://github.com/canonical/zookeeper-operator/commit/9ebd9a2050e0bd626feb0019222d45f211ca7774)           | 
```

File: docs/reference/releases/rev156_126.md [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+|------------------------|--------------|------------------|---------------------------------------------------------------------------------------------------------------------|
+| zookeeper distribution | 3.x          | 3.8.2-ubuntu0    | [5bb82d](https://git.launchpad.net/zookeeper-releases/tree/?h=lp-3.8.2&id=5bb82df4ffba910a5b30dd42499921466405f087) |
+| kafka distribution     | 3.x          | 3.6.0-ubuntu0    | [424389](https://git.launchpad.net/kafka-releases/tree/?h=lp-3.6.0&id=424389bb8f230beaef4ccb94aca464b5d22ac310)     |
+| charmed-zookeeper-snap | 3/stable     | 28               | [9757f4](https://github.com/canonical/charmed-zookeeper-snap/tree/9757f4a2a889981275f8f2a1a87e1c78ae1adb77)         |        
+| zookeeper-operator     | 3/stable     | 126              | [9ebd9a](https://github.com/canonical/zookeeper-operator/commit/9ebd9a2050e0bd626feb0019222d45f211ca7774)           | 
+| charmed-kafka-snap     | 3/stable     | 30               | [c0ce27](https://github.com/canonical/charmed-kafka-snap/tree/c0ce275f70f688e66f10f295456d2b5ff33d4f64)             |
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:49]:
```suggestion
| `charmed-kafka-snap`     | 3/stable     | 30               | [c0ce27](https://github.com/canonical/charmed-kafka-snap/tree/c0ce275f70f688e66f10f295456d2b5ff33d4f64)             |  
```

File: docs/reference/releases/rev156_126.md [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+| zookeeper distribution | 3.x          | 3.8.2-ubuntu0    | [5bb82d](https://git.launchpad.net/zookeeper-releases/tree/?h=lp-3.8.2&id=5bb82df4ffba910a5b30dd42499921466405f087) |
+| kafka distribution     | 3.x          | 3.6.0-ubuntu0    | [424389](https://git.launchpad.net/kafka-releases/tree/?h=lp-3.6.0&id=424389bb8f230beaef4ccb94aca464b5d22ac310)     |
+| charmed-zookeeper-snap | 3/stable     | 28               | [9757f4](https://github.com/canonical/charmed-zookeeper-snap/tree/9757f4a2a889981275f8f2a1a87e1c78ae1adb77)         |        
+| zookeeper-operator     | 3/stable     | 126              | [9ebd9a](https://github.com/canonical/zookeeper-operator/commit/9ebd9a2050e0bd626feb0019222d45f211ca7774)           | 
+| charmed-kafka-snap     | 3/stable     | 30               | [c0ce27](https://github.com/canonical/charmed-kafka-snap/tree/c0ce275f70f688e66f10f295456d2b5ff33d4f64)             |  
+| kafka-operator         | 3/stable     | 156              | [01d65c](https://github.com/canonical/kafka-operator/tree/01d65c3444b593d5f18d197a6514421afd3f2bc6)                 |
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:50]:
```suggestion
| `kafka-operator`         | 3/stable     | 156              | [01d65c](https://github.com/canonical/kafka-operator/tree/01d65c3444b593d5f18d197a6514421afd3f2bc6)                 |   
```

File: docs/reference/releases/rev156_126.md [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+| zookeeper distribution | 3.x          | 3.8.2-ubuntu0    | [5bb82d](https://git.launchpad.net/zookeeper-releases/tree/?h=lp-3.8.2&id=5bb82df4ffba910a5b30dd42499921466405f087) |
+| kafka distribution     | 3.x          | 3.6.0-ubuntu0    | [424389](https://git.launchpad.net/kafka-releases/tree/?h=lp-3.6.0&id=424389bb8f230beaef4ccb94aca464b5d22ac310)     |
+| charmed-zookeeper-snap | 3/stable     | 28               | [9757f4](https://github.com/canonical/charmed-zookeeper-snap/tree/9757f4a2a889981275f8f2a1a87e1c78ae1adb77)         |        
+| zookeeper-operator     | 3/stable     | 126              | [9ebd9a](https://github.com/canonical/zookeeper-operator/commit/9ebd9a2050e0bd626feb0019222d45f211ca7774)           | 
+| charmed-kafka-snap     | 3/stable     | 30               | [c0ce27](https://github.com/canonical/charmed-kafka-snap/tree/c0ce275f70f688e66f10f295456d2b5ff33d4f64)             |  
+| kafka-operator         | 3/stable     | 156              | [01d65c](https://github.com/canonical/kafka-operator/tree/01d65c3444b593d5f18d197a6514421afd3f2bc6)                 |   
+
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:51]:
```suggestion
```

File: docs/reference/releases/rev156_126.md [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+
+## Technical notes
+
+* Kafka cluster is secured-by default, meaning that when deployed if there are no client charms related to it, external listeners will not be enabled.
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:55]:
```suggestion
* A Charmed Kafka cluster is secure by default, meaning that when deployed if there are no client charms related to it, external listeners will not be enabled.
```

File: docs/reference/releases/rev156_126.md [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+* We recommend to deploy one `data-integrator` with `extra-user-roles=admin` alongside the Kafka deployment, in order to enable listeners and also create one user with elevated permission 
+  to perform administrative tasks. See the [How-to manage application](/t/charmed-kafka-documentation-how-to-manage-app/10285) userguide for more information.
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:57]:
```suggestion
  to perform administrative tasks. For more information, see the [How-to manage application](/t/charmed-kafka-documentation-how-to-manage-app/10285) guide
```

File: docs/reference/releases/rev156_126.md [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+* The current release was tested and provides compatibility with Juju 2.9.45+ and Juju 3.1+
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:58]:
```suggestion
* The current release has been tested and provides compatibility with Juju 2.9.45+ and Juju 3.1+.
```

File: docs/reference/releases/rev156_126.md [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+* Inplace upgrade for charms tracking `latest` is not supported, both for ZooKeeper and Kafka. Perform a migration of the data to upgrade to a Kafka cluster managed via a `3/stable` charm.
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:59]:
```suggestion
* Inplace upgrade for charms tracking `latest` is not supported, both for ZooKeeper and Kafka charms. Perform data migration to upgrade to a Charmed Kafka cluster managed via a `3/stable` charm. 
```

File: docs/reference/releases/rev156_126.md [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+
+
+## Technical notes
+
+* Kafka cluster is secured-by default, meaning that when deployed if there are no client charms related to it, external listeners will not be enabled.
+* We recommend to deploy one `data-integrator` with `extra-user-roles=admin` alongside the Kafka deployment, in order to enable listeners and also create one user with elevated permission 
+  to perform administrative tasks. See the [How-to manage application](/t/charmed-kafka-documentation-how-to-manage-app/10285) userguide for more information.
+* The current release was tested and provides compatibility with Juju 2.9.45+ and Juju 3.1+
+* Inplace upgrade for charms tracking `latest` is not supported, both for ZooKeeper and Kafka. Perform a migration of the data to upgrade to a Kafka cluster managed via a `3/stable` charm. 
+  See [How-to migrate a cluster](/t/charmed-kafka-documentation-how-to-migrate-a-cluster/10951) userguide for more information on how to achieve this.
```
Comment (izmalk) [inline docs/reference/releases/rev156_126.md:60]:
```suggestion
  For more information on how to do that, see the [How-to migrate a cluster](/t/charmed-kafka-documentation-how-to-migrate-a-cluster/10951) guide. 
```

File: docs/reference/releases/rev156_136.md [hunk 21]
```diff
@@ -0,0 +1,50 @@
...
+* [DPE-3726] Workload upgrade to 3.8.4-ubuntu0
+* [DPE-4183] (backport) fix: only handle quorum removal on relation-departed
+* [DPE-4362] (backport) add alive check fix
+
+Canonical Data issues are now public on both [Jira](https://warthogs.atlassian.net/jira/software/c/projects/DPE/issues/) 
+and [GitHub](https://github.com/canonical/kafka-operator/issues) platforms.
+
+[GitHub Releases](https://github.com/canonical/kafka-operator/releases) provide a detailed list of bugfixes, PRs, and commits for each revision.
+
+## Inside the charms:
```
Comment (izmalk) [inline docs/reference/releases/rev156_136.md:23]: The idea is the same as the previous release notes. Generally speaking - we don't want a colon to be a part of a heading and a list needs an introductory sentence or at the very least some text between the list and the previous heading.

**Review by izmalk — COMMENTED** [review]
Added some suggestions on how to improve wording.

File: docs/reference/releases/rev156_136.md [hunk 22]
```diff
@@ -0,0 +1,50 @@
...
+
+Dear community,
+
+We are glad to report that we have just released a new updated version for Charmed ZooKeeper on the `3/stable` channel, upgrading its revision from 126 to 136. 
+The release of a new version was promoted by the need of backporting some features that 
+should provide increased robustness and resilience during operation as well as smaller workload upgrades and fixes. See the technical notes for further information.
+
+Please reach out should you have any question, comment, feedback or information. You can find us here in [Matrix](https://matrix.to/#/#charmhub-data-platform:ubuntu.com) or also on [Discourse](https://discourse.charmhub.io/).
+
+## Features
```
Comment (izmalk) [inline docs/reference/releases/rev156_136.md:12]: Is it `New features` maybe?

File: docs/reference/releases/rev156_136.md [hunk 23]
```diff
@@ -0,0 +1,50 @@
...
+
+* [DPE-3726] Workload upgrade to 3.8.4-ubuntu0
+* [DPE-4183] (backport) fix: only handle quorum removal on relation-departed
```
Comment (izmalk) [inline docs/reference/releases/rev156_136.md:15]: It's not a new feature, but rather a bug fix. Consider creating a bugfix section for this and the next one.

File: docs/reference/releases/rev156_136.md [hunk 24]
```diff
@@ -0,0 +1,50 @@
...
+* [DPE-4362] (backport) add alive check fix
+
+Canonical Data issues are now public on both [Jira](https://warthogs.atlassian.net/jira/software/c/projects/DPE/issues/) 
+and [GitHub](https://github.com/canonical/kafka-operator/issues) platforms.
+
+[GitHub Releases](https://github.com/canonical/kafka-operator/releases) provide a detailed list of bugfixes, PRs, and commits for each revision.
```
Comment (izmalk) [inline docs/reference/releases/rev156_136.md:21]: Basically, the GitHub Releases description contains release notes. 
What is the difference between GitHub Releases and this Release notes? I'd say they should be pretty much identical.

File: docs/reference/releases/rev156_126.md (hunk 1 above)
Comment (deusebio) [reply docs/reference/releases/rev156_126.md:6]: I just did copy and paste of the annoucements we did in the public channel. I'll keep the phrasing as it was the original one.

File: docs/reference/releases/rev156_126.md (hunk 2 above)
Comment (deusebio) [reply docs/reference/releases/rev156_126.md:10]: as above

File: docs/reference/releases/rev156_126.md (hunk 3 above)
Comment (deusebio) [reply docs/reference/releases/rev156_126.md:12]: as above

**Review by izmalk — APPROVED** [review]
I'm not exactly sure what I can suggest to change here if we are just reposting existing announcements. The template for such release notes is under development now. Let's leave the text as it is right now. 
I hope to be able to help with the texts for new releases.

**Comment by deusebio** [comment]
The changes have been propagated to Discourse. Gatekeeper will update the current open PR

**Pull request closed without merging**